File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;

    int a = n;
    int b = m;
    if(n<m){a = m;b=n;}
    int r = a%b;
    
    while(r){
        if(b>r){a = b;b=r;}
        else{a=r;}

        r = a%b;     
    }
    
    answer.push_back(b);
    answer.push_back(b*(m/b)*(n/b));
    
    return answer;

}

File: 프로그래머스/1/12940.최대공약수와 최소공배수-Python3/최대공약수와 최소공배수.py
def gcd(a, b):
    if b > a:
        b,a = a,b
    while a % b != 0:
        a = a % b
        if b > a:
            b, a = a, b
    return b

def solution(n, m):
    g = gcd(n, m)
    answer = [g, (n//g)*(m//g)*g]
    return answer